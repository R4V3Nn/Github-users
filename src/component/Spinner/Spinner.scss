.loader {
	display: flex;
	height: 100px;
  }
  
  .circle {
	
	width: 40px;
	height: 40px;
	margin: 0 5px;
	border-radius: 50px;
	animation: bounce 1s ease-in infinite;
	transition: background-color 0.5s ease;
  }

  .circle:nth-child(2) {
	  animation-delay: 0.15s;
  }

  .circle:nth-child(3) {
	animation-delay: 0.3s;
}

  @keyframes bounce {
	0% {
		transform: translateY(0px);
		background-color: rgb(128, 230, 128);
	}
	50% {
		transform: translateY(-75px);
	}
	100% {
		transform: translateY(0px);
		background-color: rgb(139, 139, 235);
	}
  }