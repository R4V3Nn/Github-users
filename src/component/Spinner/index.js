import React from 'react';
import './Spinner.scss';

const Spinner = () => (
	<div className="spinner">
		<div className="loader">
			<div className="circle" />
			<div className="circle" />
			<div className="circle" />
		</div>
	</div>
);

export default Spinner;
