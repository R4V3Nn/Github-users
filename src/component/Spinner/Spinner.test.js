import React from 'react';
import { shallow } from 'enzyme';

import Spinner from './';

describe('Spinner component', () => {
	it('should render Spinner component', () => {
		// when
		const wrapper = shallow(<Spinner />);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
