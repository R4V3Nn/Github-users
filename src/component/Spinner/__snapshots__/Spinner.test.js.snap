// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render Spinner component 1`] = `
<div
  className="spinner"
>
  <div
    className="loader"
  >
    <div
      className="circle"
    />
    <div
      className="circle"
    />
    <div
      className="circle"
    />
  </div>
</div>
`;
